projects/igniteui-angular/src/lib/services/overlay/overlay.ts

```diff
@@ -542,29 +558,65 @@ export class IgxOverlayService implements OnDestroy {
         }
     }
 
-    private getOverlayInfo(component: any, moduleRef?: { injector: Injector, componentFactoryResolver: ComponentFactoryResolver }): OverlayInfo {
+    private getUserOverlaySettings(
+      viewContainerRefOrSettings: ViewContainerRef | OverlaySettings | undefined,
+      moduleRefOrSettings?: { injector: Injector, componentFactoryResolver: ComponentFactoryResolver } | OverlaySettings): OverlaySettings | undefined {
+        let result: OverlaySettings | undefined;
+        if (viewContainerRefOrSettings && !(viewContainerRefOrSettings instanceof ViewContainerRef)) {
+            result = viewContainerRefOrSettings;
+            return result;
+        }
+        if (moduleRefOrSettings && !('injector' in moduleRefOrSettings && 'componentFactoryResolver' in moduleRefOrSettings)) {
+            result = moduleRefOrSettings;
+        }
+        return result;
+    }
+
+
+    private getUserViewContainerOrModuleRef(
+        viewContainerRefOrSettings: ViewContainerRef | OverlaySettings | undefined,
+        moduleRefOrSettings?: { injector: Injector, componentFactoryResolver: ComponentFactoryResolver } | OverlaySettings
```

Comment:
I see this being used a lot so maybe we should extract it to an interface / type to keep repetitions to a minimum.

Reply:
This is correct, but still no need to do it. `ComponentFactoryResolver` is deprecated and we will remove `moduleRefOrSettings` in some of the next major releases.